multi-tenant-configuration/configure_groups.py

```diff
@@ -0,0 +1,454 @@
+    :type existing_group: dict
+    :param tenant_id: The target tenant
+    :type tenant_id: str
+    """
+    log(f"Check roles for group {group['name']}.")
+
+    group_roles = __extract_roles_from_group(group=group, tenant_id=tenant_id)
+    existing_group_roles = sorted(existing_group['roles'].split(","))
+    log("Config group roles: ", group_roles)
+
+    roles = existing_group_roles.copy()
+    missing_roles = [role for role in group_roles if role not in existing_group_roles]
+    additional_roles = [role for role in existing_group_roles if role not in group_roles]
+
+    if group_roles == existing_group_roles:
+        log('Group roles match.')
+    else:
+        print("Existing group roles: ", existing_group_roles)
+        if missing_roles:
+            print("Missing roles: ", missing_roles)
+            action_allowed = check_or_ask_for_permission(
+                target_type='group',
+                action='add missing group roles',
+                target_name=group['name'],
+                tenant_id=tenant_id,
+                option_i=True
+            )
+            if action_allowed == 'i':
+                for role in missing_roles:
+                    action_allowed = get_yes_no_answer(f"Add role {role} to group {group['name']}?")
+                    if action_allowed:
+                        roles.append(role)
+            elif action_allowed:
+                for role in missing_roles:
+                    roles.append(role)
+
+        if additional_roles:
+            print("Additional roles: ", additional_roles)
+            action_allowed = check_or_ask_for_permission(
+                target_type='group',
+                action='remove additional group roles',
+                target_name=group['name'],
+                tenant_id=tenant_id,
+                option_i=True
+            )
+            if action_allowed == 'i':
+                for role in additional_roles:
+                    action_allowed = get_yes_no_answer(f"remove role {role} from group {group['name']}?")
+                    if action_allowed:
+                        roles.remove(role)
+            elif action_allowed:
+                for role in additional_roles:
+                    roles.remove(role)
+
+        if roles != existing_group_roles:
+            # roles = ",".join(list(dict.fromkeys(roles)))
+            roles = ",".join(roles)
+            update_group(tenant_id, group, overwrite_roles=roles)
+
+
+def get_group(group: dict, tenant_id: str):
+    """
+    Checks if the group exists on the specified tenant
+    :param group: The group as defined in the configuration file
+    :type group: dict
+    :param tenant_id: The target tenant
+    :type tenant_id: str
+    :return: The group as specified on the tenant if it exists or False
+    """
+    log(f"check if group {group['name']} exists.")
+
+    url = f"{CONFIG.tenant_urls[tenant_id]}/api/groups/{group['identifier']}"
```

Comment:
Now I finally realized why I think that looks weird - I'd prefer String.format with {} placeholders.